emcc.py

```diff
@@ -543,13 +540,15 @@ def get_all_js_library_funcs(temp_files):
       if shared.is_c_symbol(name):
         name = shared.demangle_c_symbol_name(name)
         library_fns_list.append(name)
-        # TODO(sbc): wasm-ld shouldn't be reporting errors for symbols
-        # such as __wasi_fd_write which are defined with import_name attibutes
-        # but it currently does.  Remove this once we fix wasm-ld.
-        library_fns_list.append('__wasi_' + name)
   finally:
+    shared.Settings.ONLY_CALC_JS_SYMBOLS = False
     shared.Settings.INCLUDE_FULL_LIBRARY = old_full
-    shared.Settings.LINKABLE = old_linkable
+
+  # Some of these functions are actually not JS functions at all but fallbacks
+  # missing undefined native symbols.  Ignore these.
```

Comment:
`fallbacks [to]`?

But why are these special? Is this just fastcomp?

Reply:
I think the problem is that sometimes these sybmols were coming from native code and sometimes from JS.  I'll try to revert this part for now as it was only causes a few tests to fail IIRC.